assets/sass/base.scss

```diff
@@ -12,6 +12,11 @@ body {
   background: $color-neutral--lightest;
   color: $color-neutral--darkest;
   font-family: $font-family-content;
+  height: 100%;
+
+  &.--sticky {
```

Comment:
This would then also be `&.has-sticky-header`.

Maybe you can extract this to a “component” of sorts. If you give the `body` a class of `.body` and create a `body.scss`, this “base”-stylesheet does not need to know about special cases.